inc/namespace.php

```diff
@@ -379,13 +379,22 @@ function log_elasticsearch_request_errors( $response, string $context, string $c
 	$request_response_code = (int) wp_remote_retrieve_response_code( $response );
 	$is_valid_res = ( $request_response_code >= 200 && $request_response_code <= 299 );
 
+	if ( $is_valid_res ) {
+		return;
+	}
+
 	if ( is_wp_error( $response ) ) {
 		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		trigger_error( sprintf( 'Error in ElasticSearch request: %s (%s)', $response->get_error_message(), $response->get_error_code() ), E_USER_WARNING );
+	} elseif ( ! $request_response_code ) {
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		trigger_error( sprintf( 'Unknown error in ElasticSearch request: %s', wp_json_encode( $response ) ), E_USER_WARNING );
```

Comment:
Not entirely sure what's the best way to go about this, any better ideas ? @roborourke

Reply:
Ideally need to take some samples of the outputs provided at this stage. I'm not sure why some would be empty but perhaps the response code and headers would be useful to see 🤔

Reply:
This is a sample of the output from the actual client implementation:

```
Warning[18-Jan-2022 11:35:06 UTC] PHP Warning: Unexpected error in ElasticSearch request: {"headers":{},"body":"","response":{"code":false,"message":""},"cookies":[],"filename":null,"http_response":{"data":null,"headers":null,"status":null}}
```

Reply:
oh right, so the request itself is empty... regarding indexing I've seen this happen because in the `prepare_document()` function of an `Indexable` subclass it can return `false` if the post isn't found for instance.

I only saw this when a site had an unclosed `switch_to_blog()` happening in some post content output. Note that the `the_content` filter is run when processing posts for indexing to get the full content.

Reply:
It's rather the response that's empty, not the request.

Reply:
Is it not the full query? That object has `headers` and `body` properties then also `response` and `http_response` properties. 🤦‍♂️  - just remembered as I'm writing this that's how WP structures the response data.